### DISCUSSION TITLE
Deployment and testing in same workflow?

### DISCUSSION THREAD

**Discussion by pvandyken** [khanlab/snakebids#80] [General]
Just a small point of confusion for me. Currently, linting, testing and deploying are all in a single workflow that runs on every merge and pull request. Admittedly, deploy only runs when a tag is pushed, but doesn't that mean that if a PR with a tag is pushed, it will immediately publish the PR? 

Perhaps deployment should be moved to its own workflow?

**Comment by akhanf** [comment]
Yea makes more sense to have it's own workflow I think. 
Right now I think if any commit with a tag starting with 'v' is pushed it will run the deployment part..

**Comment by pvandyken** [comment]
Not totally sure yet what @kaitj  has planned with the release drafter, but I think part of the idea is to deploy when a new release is published on Github? So would we make a new workflow that runs in response to release publication?

**Reply by kaitj** [reply]
Not sure how I missed this! Yes, we would have a new workflow that runs when there is a release and it has a `v*.*.*` tag. I have something drafted for this, but waiting to merge in what we have for the `release-drafter` first.

**Reply by pvandyken** [reply]
No worries, haha. I wasn't sure if Discussion was an ideal place to leave things, so I partially wanted to test the waters with this question
